### PR TITLE
fix: hidden columns not showing after promotion to moderator

### PR DESCRIPTION
**Pull request by BenedictHomuth**
## Description

Whenever someone gets promoted to moderator, he/she should be able to see the hidden columns. This is now possible since we sync columns (`ColumnsUpdated`) on a `ParticipantUpdated` event. 

Before, this call was not present resulting in a situation where the newly named moderator needed to refresh the page to finally see the hidden columns.

Since #4179 has broadened the scope of a BoardSession DTO – specifically the AccountType – I needed to extend our DB calls, since when I parse the event in the EventFilter, it errors, since AccountType can not be an empty string (at least for now).

Therefore we should check and decide, if we want to publish the AccountType information or not. If not, I can also just annotate the AccountType in the DTO with "omitempty".

## Changelog

- added the accounttype to our db calls 
- added caching to board participants in the event filter
- added the column refresh for when a participant got updated.
## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] The light- and dark-theme are both supported and tested
- [ ] The design was implemented and is responsive for all devices and screen sizes
- [ ] The application was tested in the most commonly used browsers (e.g. Chrome, Firefox, Safari)

### CODE REVIEW

**Review by Schwehn42 — COMMENTED** [review]
> Therefore we should check and decide, if we want to publish the AccountType information or not. If not, I can also just annotate the AccountType in the DTO with "omitempty".

`AccountType` is required for #4229, does this have an impact on it?

**Comment by BenedictHomuth** [comment]
> > Therefore we should check and decide, if we want to publish the AccountType information or not. If not, I can also just annotate the AccountType in the DTO with "omitempty".
> 
> `AccountType` is required for #4229, does this have an impact on it?

If I see it correctly, the information of `AccountType` is planned to be available in the frontend. Therefore I would suggest keeping this PR as is, since with the changes present in this PR the `AccountType` is now published – at least from a backend perspective.

File: server/src/services/boards/sessions.go [hunk 1]
```diff
@@ -289,6 +289,20 @@ func (s *BoardSessionService) UpdatedSession(board uuid.UUID, session database.B
...
+	}
+
+	err = s.realtime.BroadcastToBoard(board, realtime.BoardEvent{
+		Type: realtime.BoardEventColumnsUpdated,
+		Data: dto.Columns(columns),
+	})
+	if err != nil {
+		logger.Get().Errorw("unable to broadcast update columns following a updatedsession", "err", err)
+	}
+
```
Comment (mateo-ivc) [inline server/src/services/boards/sessions.go:305]: Have you checked if the notes in the hidden column are displayed when a user is promoted to moderator?
As far as I know, BoardEventColumnsUpdated only sends information about the columns, but not about the notes in them

Comment (BenedictHomuth) [reply server/src/services/boards/sessions.go]: Good catch! 🤝 

Seems that it was to late yesterday 😅 
Fixed it!

**Review by mateo-ivc — APPROVED** [review]
lgtm

**Pull request merged**